flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestValuesTableFactory.java

```diff
@@ -89,8 +89,11 @@
 import static org.apache.flink.util.Preconditions.checkArgument;
 
 /**
- * Test implementation of {@link DynamicTableSourceFactory} that creates
- * a source that produces a sequence of values.
+ * Test implementation of {@link DynamicTableSourceFactory} that creates a source that produces a sequence of values.
+ * And this source {@link TestValuesTableSource} supports FilterPushDown. And it has some limitations.
```

Comment:
And {@link TestValuesTableSource} can push down filter into table source. A predicate can be pushed down only if it satisfies the following conditions:
1. field name is in `filterable-fields` which is defined in `with` properties
2. the field type is comparable
3. UDF is `UPPER` or `LOWER`

Reply:
thanks